router/state/certchecker.go

```diff
@@ -57,18 +65,24 @@ type CertExpirationChecker struct {
 	extender CertExtender
 }
 
-func NewCertExpirationChecker(id *identity.TokenId, edgeConfig *routerEnv.EdgeConfig, ctrls routerEnv.NetworkControllers, closeNotify <-chan struct{}) *CertExpirationChecker {
+func NewCertExpirationChecker(env CertEnv) *CertExpirationChecker {
```

Comment:
It appears the core of the work is not to depend on routerEnv.EdgeConfig and the rest is just converting to a monad;